pkg/neorpc/result/peers.go

```diff
@@ -50,11 +52,47 @@ func (g *GetPeers) AddBad(addrs []string) {
 func (p *Peers) addPeers(addrs []string) {
 	for i := range addrs {
 		addressParts := strings.Split(addrs[i], ":")
+		port, err := strconv.ParseUint(addressParts[1], 10, 16)
```

Comment:
BTW, if we're working with IPv6 then we probably need to adjust this code as it was done with https://github.com/nspcc-dev/neo-go/pull/2827. @roman-khimov?

Reply:
Technically `AddressAndTime` in P2P has 16 bytes for IP, but no one has ever used that I think. But we better future-proof it now, yes.

Reply:
What exactly should I do? That PR is pretty big and I still see this kind logic over all the place there
```
lastPort, err := strconv.ParseUint(addrStr[lastCln+1:], 10, 16)
```

Reply:
My concern was about the `addressParts[1]` line. The code you've mentioned will handle the case of some `2001:DB0:0:123A::30` address, and the following code won't:
```
addressParts := strings.Split(addrs[i], ":")
port, err := strconv.ParseUint(addressParts[1], 10, 16)
```
However, we may create a separate issue for that, @roman-khimov?

Reply:
Even though this issue has always existed there (old code doing simple split), I'd rather solve it in this PR (it's relatively easy to fix), would be a nice improvement.

Reply:
Using this should be fine, no? https://pkg.go.dev/net#SplitHostPort
It won't work for `2001:DB0:0:123A::30` but if I'm not mistaken it will never have that formatbut will look like `[2001:DB0:0:123A]::30` that does work.

That assumes this is the only place that feeds the `addrs []string` argument
https://github.com/nspcc-dev/neo-go/blob/d54418bad076057edcf24ee44876d6467d1571a6/pkg/network/tcp_peer.go#L417-L422

Reply:
Yes, it should work, let's use the https://pkg.go.dev/net#SplitHostPort.